File: src/medium/1dfroggereasy.py
n, s, m = [int(d) for d in input().split()]
b = [int(d) for d in input().split()]
visited, hops = set(), 0
while True:
    if s in visited:
        print("cycle")
        break
    elif s < 1:
        print("left")
        break
    elif s > n:
        print("right")
        break
    elif b[s - 1] == m:
        print("magic")
        break
    else:
        visited.add(s)
        s += b[s - 1]
        hops += 1
print(hops)

File: src/1dfroggereasy.cpp
#include <iostream>

using namespace std;

int main()
{
    int n, s, m, curspace, cnt = 0;
    cin >> n >> s >> m;
    int board[n];

    for (int i = 0; i < n; i++)
    {
        int tmp;
        cin >> tmp;
        board[i] = tmp;
    }
    curspace = s - 1;

    while (true)
    {
        if (curspace < 0)
        {
            cout << "left" << endl;
            break;
        }
        else if (curspace > n - 1)
        {
            cout << "right" << endl;
            break;
        }
        int temp = board[curspace];
        if (temp == 999)
        {
            cout << "cycle" << endl;
            break;
        }
        else if (temp == m)
        {
            cout << "magic" << endl;
            break;
        }
        else
        {
            cnt += 1;
            board[curspace] = 999;
            curspace += temp;
        }
    }
    cout << cnt << endl;
}
